File: app/src/main/java/com/jgm90/cloudmusic/utils/AnimationUtils.java
package com.jgm90.cloudmusic.utils;

import android.animation.ArgbEvaluator;
import android.animation.ValueAnimator;
import android.view.View;

public class AnimationUtils {

    public static void animColorChange(final View view, int color1, int color2) {
        ValueAnimator anim = new ValueAnimator();
        anim.setIntValues(color1, color2);
        anim.setEvaluator(new ArgbEvaluator());
        anim.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator valueAnimator) {
                view.setBackgroundColor((Integer) valueAnimator.getAnimatedValue());
            }
        });
        anim.setDuration(300);
        anim.start();
    }
}

File: app/src/main/java/com/jgm90/cloudmusic/core/ui/animation/AnimationUtils.kt
package com.jgm90.cloudmusic.core.ui.animation

import android.animation.ArgbEvaluator
import android.animation.ValueAnimator
import android.view.View

object AnimationUtils {
    @JvmStatic
    fun animColorChange(view: View, color1: Int, color2: Int) {
        val anim = ValueAnimator()
        anim.setIntValues(color1, color2)
        anim.setEvaluator(ArgbEvaluator())
        anim.addUpdateListener { valueAnimator ->
            view.setBackgroundColor(valueAnimator.animatedValue as Int)
        }
        anim.duration = 300
        anim.start()
    }
}
